network/src/main/java/com/pokedexcompose/network/dsl/KtorClientDsl.kt

```diff
@@ -0,0 +1,82 @@
+package com.pokedexcompose.network.dsl
+
+import com.pokedexcompose.network.dsl.model.KtorClientDslModel
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.client.request.host
+import io.ktor.client.request.parameter
+import io.ktor.client.request.port
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.HttpMethod
+import io.ktor.http.URLBuilder
+import io.ktor.http.appendPathSegments
+import io.ktor.http.contentType
+import java.lang.Exception
+
+@DslMarker
+annotation class KtorClientDsl
+
+@KtorClientDsl
+suspend inline fun <reified T, reified Body> HttpClient.request(
+    setup: KtorClientDslModel<Body>.() -> Unit
+): T {
+    val configuration = KtorClientDslModel<Body>()
+    configuration.setup()
+
+    return when (configuration.method) {
+        HttpMethod.Get -> getRequest<Body, T>(configuration)
+
+        HttpMethod.Post -> postRequest<Body, T>(configuration)
+
+        else -> getRequest<Body, T>(configuration)
+    }
+}
+
+suspend inline fun <reified Body, reified T> HttpClient.postRequest(
+    configuration: KtorClientDslModel<Body>
+): T =
+    this.post(configuration.url) {
+        contentType(configuration.contentType)
+        url { pathSegments(configuration.pathSegments) }
+        parameters(configuration.parameters)
+        headers(configuration.headers)
+        if (configuration.host.isNotEmpty()) host = configuration.host
+        if (configuration.port != -1) port = configuration.port
+        configuration.body?.let { setBody(it) }
+    }.body()
+
+suspend inline fun <reified Body, reified T> HttpClient.getRequest(
+    configuration: KtorClientDslModel<Body>
+): T =
+    this.get(configuration.url) {
+        url { pathSegments(configuration.pathSegments) }
+        parameters(configuration.parameters)
+        headers(configuration.headers)
+        if (configuration.host.isNotEmpty()) host = configuration.host
+        if (configuration.port != -1) port = configuration.port
+    }.body()
```

Comment:
Boa parte do codio de get e post é igual, e se entendi bem , é quase uma configuração basica pra funcionamento.
Sendo assim, ficaa sugestão de criar uma extension para fazer extrair essa codigo.
```suggestion
suspend inline fun <reified Body, reified T> HttpClient.postRequest(
    configuration: KtorClientDslModel<Body>
): T =
    this.post(configuration.url) {
        contentType(configuration.contentType)
        configuration.body?.let { setBody(it) }
        configBasicSetting<Body>(configuration)
    }.body()

suspend inline fun <reified Body, reified T> HttpClient.getRequest(
    configuration: KtorClientDslModel<Body>
): T =
    this.get(configuration.url) {
        url { pathSegments(configuration.pathSegments) }
        configBasicSetting<Body>(configuration)
    }.body()

inline fun <reified Body> HttpRequestBuilder.configBasicSetting(configuration: KtorClientDslModel<Body>) {
    url { pathSegments(configuration.pathSegments) }
    parameters(configuration.parameters)
    headers(configuration.headers)
    if (configuration.host.isNotEmpty()) host = configuration.host
    if (configuration.port != -1) port = configuration.port
}
```